ArjunaJTS/jts/classes/com/arjuna/ats/internal/jts/orbspecific/coordinator/ArjunaTransactionImple.java

```diff
@@ -500,6 +500,9 @@ public void rollback () throws SystemException
 			destroyAction();
 
 			status = super.status();
+		} else if ((status == ActionStatus.ABORTING)
+				|| (status == ActionStatus.ABORTED) || (status == ActionStatus.H_ROLLBACK)) {
+			return;
```

Comment:
Since you will need to fix the test below what do you think about adding an info message informing the user that we have hit this race condition:

```java
			if (jtsLogger.logger.isInfoEnabled()) {
				jtsLogger.i18NLogger.info_rollback_aborted_transaction(get_transaction_name());
			}
```

Where you will need to add the the info_rollback_aborted_transaction method to our logger com.arjuna.ats.jts.logging.jtsI18NLogger by adding to the end of the file (just before the final comment):

```java
	@Message(id = 22263, value = "rollback for {0} was already rolled back", format = MESSAGE_FORMAT)
	@LogMessage(level = INFO)
	public String info_rollback_aborted_transaction(String arg0);
```